README.md

```diff
@@ -1,6 +1,102 @@
-# envaridator
+# Envaridator
 
-Envaridator is a tiny environment variable validation library. It's main purpose is
-parsing, registering and validating environment variables.
+Envaridator is a really small environment variable validation library.
 
-Check `examples/` for example usage.
\ No newline at end of file
+## Install
+
+Installing the package can be achieved by:
+
+```
+yarn add envaridator
+
+npm i --save envaridator
+```
+
+## Basic usage:
+
+```
+import { Envaridator } from "envaridator";
+
+let envaridator = new Envaridator();
+
+import * as toi from "@toi/toi";
+import * as toix from "@toi/toix";
+
+process.env["CORRECT_URL"] = "https://google.com";
+const okURL = envaridator.register(
+  "CORRECT_URL",
+  toi.required().and(
+    toix.str.url({
+      protocol: "https:"
+    })
+  ),
+  "description"
+);
+
+process.env["WRONG_URL"] = "httttps://google.com";
+const badURL = envaridator.register(
+  "WRONG_URL",
+  toi.required().and(
+    toix.str.url({
+      protocol: "https:"
+    })
+  ),
+  "description for WRONG_URL"
+); // SHOULD FAIL
+
+try {
+  envaridator.validate();
+} catch (err) {
+  console.log(err.message);
+}
+```
+
+This would result in a status report:
+
+```
+The following environment variables are invalid:
+
+WRONG_URL - Invalid protocol: https:
+```
+
+## Documentation
+
+### Envaridator
+
+The get you up and running, you need an instance of `Envaridator`.
+
+`let envaridator = new Envaridator();`
+
+`Envaridator` offers few methods:
+
+- `envaridator.register(<VARIABLE_NAME>, <VALIDATOR>, <DESCRIPTION>)`
+
+The `VARIABLE_NAME` should be the actual environment variable name, the `VALIDATOR` should be a validator
+of type `Validator<T> = (value: string | undefined) => T;` and the `DESCRIPTION` is the
+description of the environment variable (usually what it is or what it does).
+
+- `envaridator.describeAll()`
+
+Returns the name and the description of the registered (in envaridator) environment variables.
+
+- `envaridator.validate()` - validates the registered variables, if one or more variables
+fail the validation - throws an error with a status report which includes the name of environment variable that failed the validation and also the reason.
+
+### Envar
+
+A successfully registered variable will return an `Envar`.
+
+The Envar object contains the description of the registered environment variable, it's value
+and it's name.
+
+For example:
+
+```export const okURL = envaridator.register('CORRECT_URL', toi.required().and(toix.str.url({ protocol: "https:" })), "description for CORRECT_URL")```
+
+- `okURL.description => "description for CORRECT_URL"`
+- `okURL.value => an URL object`
```

Comment:
I would also add that we validate the value before returning it. This way we make sure that it gets validated no matter what.

Reply:
We don't necessary need to call `envaridator.validate()` before using the `Envar` properties, since `envaridator.register()` will return an `Envar` object no matter what, and when calling the `value` property - it will validate that specific `Envar`.

But I guess it is a good idea that we should write a NOTE that says that it is a good practice to validate the registered variables before using them?

Reply:
You didn't get me I guess. I just meant that we should point out the fact that we do validate the `Envar` when `.value` is called, so the user knows that they shouldn't validate again.

Reply:
Aha! Yep, fair point. On it.

Reply:
Done.